client/components/social-buttons/apple.js

```diff
@@ -21,25 +22,99 @@ import requestExternalAccess from 'lib/sharing';
 import './style.scss';
 import AppleIcon from 'components/social-icons/apple';
 
-const connectUrl =
+const connectUrlPopupFLow =
 	'https://public-api.wordpress.com/connect/?magic=keyring&service=apple&action=request&for=connect';
 
 class AppleLoginButton extends Component {
 	static propTypes = {
+		clientId: PropTypes.string.isRequired,
 		isFormDisabled: PropTypes.bool,
+		redirectUri: PropTypes.string,
 		responseHandler: PropTypes.func.isRequired,
+		scope: PropTypes.string,
+		uxMode: PropTypes.oneOf( [ 'redirect', 'popup' ] ),
+		socialServiceResponse: PropTypes.object,
 	};
 
 	static defaultProps = {
 		onClick: noop,
+		scope: 'name email',
+		uxMode: 'popup',
 	};
 
+	componentDidMount() {
+		if ( ! config.isEnabled( 'sign-in-with-apple' ) ) {
+			return;
+		}
+
+		if ( this.props.uxMode === 'redirect' ) {
+			// do not load the client for the popup flow as
+			// we won't have a valid redirectUri to give to the apple client
+			this.loadAppleClient();
+
+			if (
+				this.props.socialServiceResponse &&
+				this.props.socialServiceResponse.client_id === config( 'apple_oauth_client_id' )
+			) {
+				const user = {
+					email: this.props.socialServiceResponse.user_email,
+					name: this.props.socialServiceResponse.user_name,
+				};
+				this.props.responseHandler( {
+					id_token: this.props.socialServiceResponse.id_token,
+					user: user,
+				} );
+			}
+		}
+	}
+
+	async loadDependency() {
+		if ( ! window.AppleID ) {
+			await loadScript(
+				'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js'
+			);
+		}
+		return window.AppleID;
+	}
+
+	loadAppleClient() {
+		if ( this.appleClientLoaded ) {
+			return this.appleClientLoaded;
+		}
+
+		this.appleClientLoaded = this.loadDependency()
+			.then( AppleID =>
+				AppleID.auth.init( {
+					clientId: this.props.clientId,
+					scope: this.props.scope,
+					redirectURI: this.props.redirectUri,
+					state: '1',
```

Comment:
Need to see if we need to set a state to prevent CSRF

Reply:
Here https://auth0.com/docs/protocols/oauth2/mitigate-csrf-attacks they recommend using the localStorage for the state in the case of a SPA. I might just do that 👍